Sprint 19

h1node opened the pull request:
- made fragments instead of Activity
 - BottomNavigationView
 - NavHostFragment

h1node committed:
Sprint 19
 - made fragments instead of Activity
 - BottomNavigationView
 - NavHostFragment

ArturNurtdinov commented on app/src/main/java/com/playlistmaker/presentation/ui/player/AudioPlayerFragment.kt:
```diff
@@ -1,57 +1,63 @@
     private fun displayTrackDetails() {
-        val track = intent?.getParcelableExtra<Music>("track") ?: return
+        arguments
+        val track = requireArguments().getParcelable<Music>("track") ?: return
```
Ключи для передачи данных лучше вынести в константу, т.к. они используются в более чем 1 месте. Ещё один плюс вынесения в константы - в будущем легче будет этот экран переиспользовать

ArturNurtdinov commented on app/src/main/java/com/playlistmaker/presentation/ui/player/AudioPlayerFragment.kt:
```diff
@@ -123,7 +129,7 @@ class AudioPlayerActivity : AppCompatActivity() {
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         if (item.itemId == android.R.id.home) {
-            finish()
+            requireActivity()
```
В данном случае семантически неверное преобразование - до этого был вызов `finish`, вероятно ты его забыл

ArturNurtdinov reviewed (commented):
1. Нижняя панель навигации отображается на экране аудиоплеера. Ожидаемое поведение - панель скрыта на экране аудиоплеера:
![image](https://github.com/user-attachments/assets/ed15e8c6-ac20-4e4b-9355-e064f2f823a4)
2. История поиска не отображается при пустом поле ввода - вместо неё отображается результат последнего поискового запроса:
![image](https://github.com/user-attachments/assets/a74a9e31-f362-490f-bcc7-1110819dd5af)
3. Кнопка "Назад" на экране аудиоплеера не работает

h1node committed:
Sprint 19
 - панель навигации убрал
 - историю поиска вроде поправил
 - кнопка назад теперь работает

ArturNurtdinov commented on app/src/main/java/com/playlistmaker/presentation/ui/player/AudioPlayerFragment.kt:
```diff
@@ -46,29 +46,44 @@ class AudioPlayerFragment : Fragment() {
         }
     }
 
+    companion object {
```
`companion object` принято объявлять внизу класса согласно Kotlin Coding Convention (https://kotlinlang.org/docs/coding-conventions.html#class-layout)
